mlrun/artifacts/manager.py

```diff
@@ -200,8 +200,8 @@ def log_artifact(
                 # and receive back all the runs that are associated with his search result.
                 db_key = producer.name + "_" + key
             else:
-                db_key = key
-        item.db_key = db_key if db_key else ""
+                db_key = item.db_key or key
```

Comment:
mind explaining the flows and why using key vs db_key would make sense in that regard?

Reply:
The `db_key` variable determines what `key` the artifact is saved with in the DB.

When you log an artifact without specifying a `db_key` explicitly, `item.db_key` will be empty - so we want to take the `key` and use that as the actual key.
If an explicit `db_key` was set, we want to use that.

This change mainly affects imported artifacts that have an explicit `db_key` value already set (either the same as `key` or a different value).
If it is set - we don't want to override it and we want to keep the same behavior.